src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -491,7 +491,7 @@ private bool ReplacementBreaksAnonymousObjectMemberDeclarator(AnonymousObjectMem
         {
             var originalExpressionType = this.OriginalSemanticModel.GetTypeInfo(originalAnonymousObjectMemberDeclarator.Expression, this.CancellationToken).Type;
             var newExpressionType = this.SpeculativeSemanticModel.GetTypeInfo(replacedAnonymousObjectMemberDeclarator.Expression, this.CancellationToken).Type;
-            return originalExpressionType != newExpressionType;
+            return !originalExpressionType.Equals(newExpressionType);
```

Comment:
1. D'oh on me if i did the original `!=`.
2. We need to do `object.Equals(original..., new...)`.  Because `originalExpressionType` can very likely be null, and that would def throw here.